InnerEye/ML/lightning_base.py

```diff
@@ -136,6 +139,27 @@ def setup(self) -> None:
         This hook reads the dataset file, and possibly sets required pre-processing objects, like one-hot encoder
         for categorical features, that need to be available before creating the model.
         """
+        # Following code validates segmentation training, validation and test data to ensure:
+        # 1) Files exist,
+        # 2) mask_id identifier is not empty,
+        # 3) consistency for input channels, ground_truth and mask_id,
+        # 4) ensures data is consistent with load_dataset_sources method prior running training, validation and testing.
+        if self.config.is_segmentation_model:
+            # Creates a list with all the channels of interest
+            all_channels = self.config.image_channels + self.config.ground_truth_ids
+            # Mask_id is an optional field. If non-empty in the config, will check dataframe.
+            if len(self.config.mask_id) > 0:
+                all_channels += [self.config.mask_id]
+            # Root directory where data is stored
+            local_dataset_root_folder = self.config.local_dataset
+            # Iterate over train, validation and test dataset
+            for split_data in [ self.config.get_dataset_splits().train, self.config.get_dataset_splits().val, self.config.get_dataset_splits().test]:
+                unique_ids = set(split_data[CSV_SUBJECT_HEADER])
+                for patient_id in unique_ids:
+                    rows = split_data.loc[split_data[CSV_SUBJECT_HEADER] == patient_id]
+                    # Validates condition so that data is consistent with
```

Comment:
... is consistent with... ?

Reply:
updated comment to: 
# Converts channels from data frame to file paths and gets errors if any